src/scheduler.h

```diff
@@ -67,23 +61,32 @@ class CScheduler
      */
     void MockForward(std::chrono::seconds delta_seconds);
 
-    // To keep things as simple as possible, there is no unschedule.
-
-    // Services the queue 'forever'. Should be run in a thread,
-    // and interrupted using boost::interrupt_thread
+    /**
+     * Services the queue 'forever'. Should be run in a thread,
+     * and interrupted using boost::interrupt_thread
+     */
     void serviceQueue();
 
-    // Tell any threads running serviceQueue to stop as soon as they're
-    // done servicing whatever task they're currently servicing (drain=false)
-    // or when there is no work left to be done (drain=true)
-    void stop(bool drain=false);
+    /** Tell any threads running serviceQueue to stop as soon as the current task is done */
+    void StopImmediately()
+    {
+        WITH_LOCK(newTaskMutex, stopRequested = true);
+        newTaskScheduled.notify_all();
+    }
+    /** Tell any threads running serviceQueue to stop when there is no work left to be done */
+    void StopWhenDrained()
+    {
+        WITH_LOCK(newTaskMutex, stopWhenEmpty = true);
+        newTaskScheduled.notify_all();
+    }
 
-    // Returns number of tasks waiting to be serviced,
-    // and first and last task times
-    size_t getQueueInfo(std::chrono::system_clock::time_point &first,
-                        std::chrono::system_clock::time_point &last) const;
+    /**
+     * Returns number of tasks waiting to be serviced,
+     * and first and last task times
+     */
+    size_t getQueueInfo(std::chrono::system_clock::time_point& first, std::chrono::system_clock::time_point& last) const;
```

Comment:
I find it more readable to to have the arguments split over multiple lines, though I guess that means I should open a PR to remove `AllowAllParametersOfDeclarationOnNextLine: false` from `.clang-format`?

Reply:
In this case splitting over multiple lines is allowed because `BinPack...` is already set to `false`. However, unless `AlignAfterOpenBracket` is changed to `Align` I find it less readable.

In any case changing `.clang-format` should be done in a separate pr, and I might object changing `AlignAfterOpenBracket` to `Align` because it wastes so much whitespace.

Reply:
https://github.com/bitcoin/bitcoin/pull/19095

Reply:
thx fixed